### PR TITLE
Add shared GitHub Actions workflow for ruby

### PR DESCRIPTION
**Pull request by archanaserver**
Used `theforeman/actions` reusable workflows for the GHA config.

https://github.com/theforeman/actions?tab=readme-ov-file#foreman-plugin-tests

Also added Gemfile and updated .rubocop.yml file to resolve error. Look for commit for more details.

### CODE REVIEW

File: .github/workflows/ruby_ci.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: CI
+
+on: pull_request
```
Comment (ekohl) [inline .github/workflows/ruby_ci.yml]: Can you change this to also test push to the develop branch?

File: Gemfile
Comment (ekohl) [inline Gemfile:1]: You need to make sure rubocop is in here. Probably easier to depend on `theforeman-rubocop`.

**Review by ekohl — CHANGES_REQUESTED** [review]
https://github.com/theforeman/foreman_default_hostgroup/pull/60 needs to be addressed before this can pass.

**Comment by archanaserver** [comment]
WIP https://github.com/theforeman/foreman_default_hostgroup/pull/63

**Comment by ekohl** [comment]
Please rebase now that #63 is merged.

**Comment by ekohl** [comment]
Could you implement what https://github.com/theforeman/foreman_default_hostgroup/pull/60#discussion_r1348314077 suggests? Otherwise it's broken.

**Comment by archanaserver** [comment]
@ekohl after making https://github.com/theforeman/foreman_default_hostgroup/pull/62/commits/d6874423adc1f91276a6690fbad853222dab48d5 changes, at this point i'm not sure what to look for to resolve this issue.

**Review by ekohl — CHANGES_REQUESTED** [review]
I think as part cbf79fe7e8a63624a39bc77d07d0e60d9166a78c we should have removed this bit:
https://github.com/theforeman/foreman_default_hostgroup/blob/c4bb39d72d98ddb96ecf7fac7f73aadff0aa45b7/test/unit/default_hostgroup_test.rb#L17-L27

Because of the settings DSL refactor and we can safely rely on the setting existing. But at the very least, drop the category bit there.

**Comment by ekohl** [comment]
I get the impression it's not patching the class, for whatever reason. It's unclear if this is also a problem outside of the tests.

**Comment by archanaserver** [comment]
@ekohl I still don't understand what I have to do here, do you have any other thoughts?

**Comment by evgeni** [comment]
> I get the impression it's not patching the class, for whatever reason. It's unclear if this is also a problem outside of the tests.

It's patching the class just fine.
The tests use the wrong class (Host::Managed, not HostFactImporter!) which didn't have a `import_facts` since Foreman 2.2 (https://github.com/theforeman/foreman/commit/5253d21ed7d690e80e595a075d06f96d2e211e8c)

**Comment by evgeni** [comment]
Well, the "real" problem is 66ad79c80321893084a1d19a26ce300ad0abb341 moved the patching from `Host` to `HostFactImporter`, but now the methods do not really make sense… IMHO

But at least they work?

**Review by ekohl — APPROVED** [review]

**Pull request merged**